docs/DeveloperGuide.md

```diff
@@ -1,34 +1,617 @@
+* _Common_: collection of classes used by multiple components.
+
+
+GULIO is a local application that stores its application data using readable text files, allowing users the flexibility of viewing and editing data locally.
+
+The way GULIO runs and handles user input can be described as follows:
+
+<p align="center">
+    <img width="973" src="developerGuideImages/main-loop-sequence-diagram.png" alt="Sequence Diagram"><br>
+    Figure 3 - GULIO Sequence Diagram
+</p>
+
+Upon start, the `Duke` (main) class calls `run()` which enters a while loop and reads in user input. In the loop, the _Parser_ component processes the user input into various commands implemented in GULIO. The loop ends when the user enters exit.
+
+&nbsp;
+
+### UI component
+
+**API**: `UI.java`
+
+* Facilitates the CLI interface
+* Methods to display general messages, prompt messages and error messages
+* Reads in user’s input, and used by Command classes to react to user’s inputs
+* The `UI` object created as an attribute in Duke is passed into each command to be executed
+* Instances of `UI` are used by tests in general
+
+&nbsp;
+
+### Parser component
+
+**API**: `Parser.java`
+
+* Determines the command entered by the user
+
+* Parses the parameters needed by subclasses of the `Command` object (for commands which require additional details)
+
+* Checks the validity of parsed parameters, in some instances calling methods from other relevant classes, e.g. calling a method from the `Lessons` class to verify parsed lesson links.
+
+* May instruct `UI` to print warnings and prompts to users, e.g. when users enter invalid parameters.
+
+* Returns a new `Command` object with all the necessary attributes filled
+
+&nbsp;
+
+### Command component
+
+<p align="center">
+    <img width="973" src="developerGuideImages/command-component-level.png" alt="Dual Layer Command System"><br>
+    Figure 4 - Visualisation of Dual Layer Command System
+</p>
+
+**API**: `Command.java`
+
+The _Command_ component consist of an abstract `Command` class, as well as its subclasses. Each subclass handles a specifc command in GULIO. The abstract `Command` class contains methods that these commands inherit from. They can be further distinguished by commands that are used when the user is at the dashboard or when the user is within a module specified.
+
+Steps for command execution:
+
+1. The `Parser` validates user input and returns a `Command` object to `Duke`
+1. `Duke` calls the execute method in the `Command` object
+1. Depending on the command, it may make changes to the objects within the _Model_ component
+1. If there are changes, the _Model_ component then updates application data via the _Storage_ component
+1. The `UI` prints user information related to the command executed
+
+
+&nbsp;
+
+### Model component
+
+<p align="center">
+    <img width="973" src="developerGuideImages/designModel.png" alt="Class Diagram of Model"><br>
+    Figure 5 - Class Diagram of Model
+</p>
+
```

Comment:
In general, the diagrams are comprehensive. However, can consider standardizing the size of each diagram